src/types/express.d.ts

```diff
@@ -0,0 +1,11 @@
+import {AppDependencies} from './app_dependencies';
+
+declare global {
+    namespace Express {
+        export interface Request {
+            deps: AppDependencies;
+        }
+    }
+}
```

Comment:
Here we decorate the request object globally, so the whole application knows about the available dependencies